runtime/criusupport/criusupport.c

```diff
@@ -352,19 +361,33 @@ Java_org_eclipse_openj9_criu_CRIUSupport_checkpointJVMImpl(JNIEnv *env,
 		if (directoryBuf != directoryChars) {
 			j9mem_free_memory(directoryChars);
 		}
-		if (NULL != currentThread->currentException) {
-			currentExceptionLocalRef = (jthrowable) vmFuncs->j9jni_createLocalRef(env, currentThread->currentException);
-			currentThread->currentException = NULL;
-
-			if (NULL == currentExceptionLocalRef) {
-				/* nothing else to do */
-				vmFuncs->setNativeOutOfMemoryError(currentThread, 0, 0);
-			}
-		}
 
 		vmFuncs->internalExitVMToJNI(currentThread);
 #endif /* defined(LINUX) */
 	}
 
-	return constructResult(env, resultType, currentExceptionLocalRef);
+	if (NULL != currentExceptionClass) {
+		msgCharLength = j9str_printf(PORTLIB, NULL, 0, nlsMsgFormat, systemReturnCode);
+		exceptionMsg = (char *)j9mem_allocate_memory(msgCharLength, J9MEM_CATEGORY_VM);
+
+		if (NULL != exceptionMsg) {
```

Comment:
If this code fails to allocate the `exceptionMsg`, we'll exit this function without setting an exception.  That seems like a mistake.  Better to pass a null string than to return without the exception pending